From d881482ca0db89000e5581565933a338ffba52dd Mon Sep 17 00:00:00 2001
From: Adrien Plazas <kekun.plazas@laposte.net>
Date: Sat, 6 Jul 2019 11:05:24 +0200
Subject: [PATCH 20/29] scrolledwindow: Set deceleration value based on the
 device

This makes it nicer on phones by avoiding having to scroll too much on a
touch screen and it makes it match the changes we applied to WebKitGTK
for the same reasons.
---
 gtk/gtkscrolledwindow.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/gtk/gtkscrolledwindow.c b/gtk/gtkscrolledwindow.c
index 95c910f..3a6192b 100644
--- a/gtk/gtkscrolledwindow.c
+++ b/gtk/gtkscrolledwindow.c
@@ -177,6 +177,7 @@
 /* Kinetic scrolling */
 #define MAX_OVERSHOOT_DISTANCE 100
 #define DECELERATION_FRICTION 4
+#define DECELERATION_FRICTION_PHONE 1
 #define OVERSHOOT_FRICTION 20
 #define SCROLL_CAPTURE_THRESHOLD_MS 150
 
@@ -3765,7 +3766,7 @@ gtk_scrolled_window_start_deceleration (GtkScrolledWindow *scrolled_window)
         gtk_kinetic_scrolling_new (lower,
                                    upper,
                                    MAX_OVERSHOOT_DISTANCE,
-                                   DECELERATION_FRICTION,
+                                   _gtk_get_is_phone () ? DECELERATION_FRICTION_PHONE : DECELERATION_FRICTION,
                                    OVERSHOOT_FRICTION,
                                    priv->unclamped_hadj_value,
                                    priv->x_velocity);
@@ -3784,7 +3785,7 @@ gtk_scrolled_window_start_deceleration (GtkScrolledWindow *scrolled_window)
         gtk_kinetic_scrolling_new (lower,
                                    upper,
                                    MAX_OVERSHOOT_DISTANCE,
-                                   DECELERATION_FRICTION,
+                                   _gtk_get_is_phone () ? DECELERATION_FRICTION_PHONE : DECELERATION_FRICTION,
                                    OVERSHOOT_FRICTION,
                                    priv->unclamped_vadj_value,
                                    priv->y_velocity);
-- 
2.29.2

